mautic/mautic
# Update Transifex dependency

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | N
| Automated tests included? | Y
| Related user documentation PR URL | /
| Related developer documentation PR URL | /
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/8301
| BC breaks? | Y
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

Check the linked issue for the why.

This PR replaces the archived library with our live fork. It also updates the code to use the latest version. This way we get rid of the dead code.

This PR also removes the Transifex credentials fields from the global config. They can be used only on development version of Mautic and only by Transifex users with admin access to the Mautic repo. People like those can edit local.php directly. The credentials in the UI were only confusing for normal users.

#### Steps to test this PR:
0. You can test this only with the GitHub version of Mautic that has the dev dependencies installed.
1. Run `composer install`
2. Try to run `bin/console mautic:transifex:push` and `bin/console mautic:transifex:pull` commands. It should give you this output:
```
Your Transifex credentials must be saved to the configuration in order to use this command.
```
That's as far as you can get and I'd consider it a successful test.

If you are a Transifex user with admin permission for the Mautic project, it will actually do the API calls. You'll see those in the command output. You can add your credentials to the app/config/local.php file like this:
```
'transifex_username' => 'your_username',
'transifex_password' => 'your_password',
```

#### List backwards compatibility breaks:
1. BabDev/Transifex/* classes were replaced with Mautic/Transifex/*
